### PR TITLE
Fix/cmem 7098 tags

### PR DESCRIPTION
**Pull request by J-x-D**

### CODE REVIEW

File: src/components/Tooltip/tooltip.scss [hunk 1]
```diff
@@ -34,6 +34,10 @@ $tooltip-padding-horizontal: $eccgui-size-block-whitespace * 0.5; // !default;
 }
 
 .#{$eccgui}-tooltip__wrapper:not(.#{$ns}-tooltip-indicator) {
+    display: flex;
```
Comment (haschek) [inline src/components/Tooltip/tooltip.scss:37]: Why are those rules necessary?

1. The content in the tooltip could get it's own wrapper with it's own styles, whatever is necessary.
2. Changing the rules of this central element leads very probably to issues on other usages of `Tooltip`.

Just in case that really a change for this class is necessary then it should added to the component stylesheet directly (tooltip folder), not somewhere else.

File: src/components/Tag/tag.scss [hunk 2]
```diff
@@ -272,6 +272,52 @@ $tag-round-adjustment: 0 !default;
     }
 }
 
+.#{$eccgui}-tag__list {
```
Comment (haschek) [inline src/components/Tag/tag.scss:275]: Most of the new rules are probably not ncessary, and maybe all of them harm the correct display by overwriting already defined rules.

File: src/components/Tag/TagList.tsx [hunk 3]
```diff
@@ -1,34 +1,215 @@
...
+                        content={
+                            <div className={`${eccgui}-tag__list-overflow-content`}>
+                                {childArray.map((child, i) => (
+                                    <React.Fragment key={"tooltip-tag-" + i}>{child}</React.Fragment>
+                                ))}
+                            </div>
+                        }
+                        size="large"
+                    >
+                        <Tag
```
Comment (haschek) [inline src/components/Tag/TagList.tsx:162]: Better to have a component property that provides a callback (that get the count of hidden items) used to render the target for the tooltip (even better, use a `ContextOverlay` because it can stay open, so the actual hidden elements could added there, including their own tooltip if they had one).

The `TagList` component then only need to add the measurements regarding hidden elements if the callback is given.

The minimal number of directly visible tags could also be another important constraint (in other contexts), and could be supported by an optional property, too.

File: src/components/Tag/TagList.tsx [hunk 4]
```diff
@@ -1,34 +1,215 @@
...
+                            +{hiddenCount} more
+                        </Tag>
+                    </Tooltip>
+                </li>
+            )}
+        </ul>
+    );
+
+    // Hidden measurement list - always rendered for measurements
+    const measurementList = (
```
Comment (haschek) [inline src/components/Tag/TagList.tsx:177]: We should not add that type of "shadow markup" only for measurement purposes, if it is not really necessary for technical reasons.

If there is the need to cut the list could probably be initially tested with `scrollWidth/Height` and `clientWidth/Height` on the correct overflow parent. Then the process could be stopped, or if there is an overflow active then use the visibility tests for the `TagList` items that you already implemented.

File: src/components/Tag/TagList.tsx [hunk 5]
```diff
@@ -1,34 +1,215 @@
...
+                    {tagList}
+                    {measurementList}
+                </span>
             </div>
         );
     }
 
-    return tagList;
+    return (
+        <div className={`${eccgui}-tag__list-container` + (className ? " " + className : "")}>
```
Comment (haschek) [inline src/components/Tag/TagList.tsx:208]: Adding another container with the additional `className` could lead to unexpect results (`className` is already added to the `ul` element if not `label` is given. The extra container may not be necessary if no extra element for the measurement is inserted.

**Review by haschek — CHANGES_REQUESTED** [review]
Additional to the requested changes, it is always necessary to update the changelog with a short comment about what was added/changed/fixed.
